src/test/java/io/confluent/connect/hdfs/ConnectSiteXMLTest.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ **/
+
+package io.confluent.connect.hdfs;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileUtil;
+import org.apache.hadoop.fs.Path;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.PrintWriter;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ConnectSiteXMLTest {
+
+  private File baseDir = new File(System.getProperty("test.build.dir", "target/test-dir"));
+
+  @Test
+  public void testConfiguration() throws Exception {
+    Configuration conf = new Configuration();
+    FileUtil.fullyDelete(baseDir);
+    assertTrue(baseDir.mkdirs());
+    createTempConnectSiteXMLFile();
+    conf.addResource(new Path(baseDir.getAbsolutePath() + "/connect-site.xml"));
+    assertEquals("connect/_HOST@EXAMPLE.COM", conf.get("connect.hdfs.principal"));
+    assertEquals("path_to_keytab", conf.get("connect.hdfs.keytab"));
+    assertEquals("kerberos", conf.get("hadoop.security.authentication"));
+    assertEquals("hdfs/_HOST@EXAMPLE.COM", conf.get("dfs.namenode.kerberos.principal"));
+    FileUtil.fullyDelete(baseDir);
+  }
+
+  private void createTempConnectSiteXMLFile() throws Exception {
+    String template = "<configuration>" +
+                        "<property>" +
+                            "<name>hadoop.security.authentication</name>" +
```

Comment:
Normally this would be in an existing config file, right? Is this just the convenient place to put it during the test to avoid creating another file?